src/device/socket/vsock.rs

```diff
@@ -223,18 +218,15 @@ impl<H: Hal, T: Transport> VirtIOSocket<H, T> {
     }
 
     fn check_peer_buffer_is_sufficient(&mut self, buffer_len: usize) -> Result {
-        if usize::try_from(self.connection_info()?.peer_free())
-            .map_err(|_| SocketError::InvalidNumber)?
-            >= buffer_len
-        {
+        let buffer_len = buffer_len
+            .try_into()
+            .map_err(|_| SocketError::InvalidNumber)?;
+        if self.connection_info()?.peer_free() >= buffer_len {
```

Comment:
I think it makes more sense to keep the conversion the way it was. Converting `u32` to `usize` should never fail except on 16-bit platforms (which we don't support anyway), whereas converting `usize` to `u32` can fail. In fact there's no need for the `try_from`, a cast should be fine.